### ISSUE TITLE
[Grid]-Drag and Drop - Not working when application is deployed on a different context than Root context

### ISSUE THREAD

**Issue by silvan-lincan** [vaadin/flow#12315]
### Description

In a Drag and Drop enabled Grid, using Grid#setRowsDraggable(true):

- if application is deployed under Root context ( http://127.0.0.1:8080 ), then Drag and Drop works both on a Desktop browser and also on a mobile device
- if application is deployed under a different context than Root ( http://127.0.0.1:8080/dnd ), then Drag and Drop works on a Desktop browser, but it is not working on a mobile device, and it fails with the following error:

<img width="1278" alt="Screenshot 2021-11-09 at 08 01 38" src="https://user-images.githubusercontent.com/47028355/140878319-0b8074a0-025f-450e-bd71-efdf774f0f22.png">

### Expected outcome

Drag and Drop among the Grid rows should work both on Desktop and Mobile, not matter if the application is deployed in the Root context or a different one.

### Actual outcome

Drag and Drop is not working on a mobile device, when the application is deployed on a different context than Root and if tails with the following error:
<img width="1278" alt="Screenshot 2021-11-09 at 08 03 28" src="https://user-images.githubusercontent.com/47028355/140878680-f2a950d1-370c-4e19-a7e6-67fc331c554f.png">

### Minimal reproducible example

The issue can be reproduced from a Vaadin 14.7.4 starter.
Please see archive attached: 
[vaadin14-dnd-mobile.zip](https://github.com/vaadin/web-components/files/7502807/vaadin14-dnd-mobile.zip)

### Steps to reproduce

1. Start the application: mvn jetty:run .
2. Open the Google Chrome Desktop browser along with the Developer Tools and set the Device to, for example "iPad Pro" (like in the screenshot). It is important to do this step first, in order to simulate a mobile device.
3. Access the application at: http://127.0.0.1:8080/dnd
4. You should obtain the error described above.

### Environment

- Vaadin 14.7.4
- Jetty: 9.4.42.v20210604 with web context set to "/dnd" => http://127.0.0.1:8080/dnd
- Google Chrome set to simulate an "iPad Pro"

### Notes
Please let me know if you need additional info
Thank you.

**Comment by silvan-lincan** [comment]
Hi @alvarezguille 

Thank you.

**Comment by silvan-lincan** [comment]
Hello,

please correct me if I am wrong, this is only an assumption as looking at the DndUtil.java, can it be that in this particular case, the wrong method is used at lines 118 and 120 and the polyfill url should be without using the "context://" protocol?

I think in this case, since these resources are in jars (outside of the frontend folder), they are treated as external urls, included in the page as standalone scripts (when needed), therefore, they are not relative to the servlet context.

I think a more appropriate call would be:
```
MOBILE_DND_POLYFILL_URL="/webjars/mobile-drag-drop/2.3.0-rc.1/index.min.js";
url1 = ui.getSession().getService().getResourceInServletContextOrWebJar(MOBILE_DND_POLYFILL_URL);
```

**Comment by Artur-** [comment]
Wow, this has been open for some time... It will be fixed by #16477
